lib/src/Uncertainty/Algorithm/MetaModel/FunctionalChaos/FunctionalChaosResult.cxx

```diff
@@ -307,6 +313,110 @@ void FunctionalChaosResult::setInvolvesModelSelection(const Bool involvesModelSe
+  
+  // Check independence
+  if (!distribution_.hasIndependentCopula())
+    throw InvalidArgumentException(HERE) << "This class can only manage an independent copula.";
+
+  // Create the conditioned orthogonal basis
+  const String basicClassName(orthogonalBasis_.getImplementation()->getClassName());
+  OrthogonalBasis orthogonalBasisMarginal;
+  if (basicClassName == "OrthogonalProductPolynomialFactory")
+  {
+    const OrthogonalProductPolynomialFactory* p_basis = dynamic_cast<const OrthogonalProductPolynomialFactory*>(orthogonalBasis_.getImplementation().get());
+    orthogonalBasisMarginal = p_basis->getMarginal(conditioningIndices);
+  }
+  else if (basicClassName != "OrthogonalProductFunctionFactory")
+  {
+    const OrthogonalProductFunctionFactory* p_basis = dynamic_cast<const OrthogonalProductFunctionFactory*>(orthogonalBasis_.getImplementation().get());
+    orthogonalBasisMarginal = p_basis->getMarginal(conditioningIndices);
+  }
+  else
+    throw InvalidArgumentException(HERE) << "This class can only manage an OrthogonalProductPolynomialFactory "
+          << "or an OrthogonalProductFunctionFactory "
+          << "but current basis is" << basicClassName;
+
+  // Create the active transformation and its inverse
+  const Distribution measureMarginal(orthogonalBasisMarginal.getMeasure());
+  const DistributionTransformation transformationMarginal(inputDistributionMarginal, measureMarginal);
+  const DistributionTransformation inverseTransformationMarginal(transformationMarginal.inverse());
+
+  // Restrict the enumeration function
+  const EnumerateFunction enumerateFunction(orthogonalBasis_.getEnumerateFunction());
+  EnumerateFunction enumerateFunctionMarginal(enumerateFunction.getMarginal(conditioningIndices));
+
+  // Condition the multi-indices (taking into account for model selection)
+  // Get the indices of active multi-indices in the reduced enumeration rule
+  Indices listOfActiveReducedIndices(0);  // In the reduced enumeration rule
+  Indices listOfActiveIndices(0);  // In the list of coefficients
+  const UnsignedInteger numberOfFunctions = Psi_k_.getSize();
+  for (UnsignedInteger k = 0; k < numberOfFunctions; ++k)
+  {
+    const Indices multiIndex(enumerateFunction(I_[k]));
+    // See if this function has active marginal indices only
+    bool isActive = true;
+    for (UnsignedInteger i = 0; i < inputDimension; ++i)
+      if (!(conditioningIndices.contains(i)) && multiIndex[i] > 0)
+      {
+        isActive = false;
+        break;
+      }
+    if (isActive)
+    {
+      listOfActiveIndices.add(k);
+      const UnsignedInteger activeDimension = conditioningIndices.getSize();
+      Indices activeMultiIndex(activeDimension);
+      UnsignedInteger localIndex = 0;
+      for (UnsignedInteger i = 0; i < inputDimension; ++i)
+        if (conditioningIndices.contains(i))
+          activeMultiIndex[localIndex] = multiIndex[i];
```

Comment:
This is a bug: `localIndex` should be increased. The bug does not reveal itself because there is no unit test with a conditioning with respect to 2 variables or more.

Reply:
Fixed at 3d3cb220f888fec6361b2fc57c9a8028dd94f102